### ISSUE TITLE
Issue with rerunning LII tests

### ISSUE THREAD

**Issue by ekluzek** [ESMCI/cime#1411]
This is a problem with cime5.3.0-alpha.02.

LII tests run two tests using compare_two_tests, when you rerun a case that's already setup, it doesn't rerun the first case. So if you fixed something and reran the test it will fail.

There weren't any issues similar to this that have been resolved on master. And I looked through changes and it's not clear if there's a fix for this. There are changes to SystemTests, so it's possible.

@billsacks

**Comment by jedwards4b** [comment]
We want to move the LII test out of cime and into CLM - there is already code in cime to handle this, we are waiting for the update on the clm side.

**Comment by billsacks** [comment]
@jedwards4b - unless you have made changes recently, we were waiting on some robustness additions before moving tests to the components - see https://github.com/ESMCI/cime/issues/651

But that's not really relevant to this issue. LII is a simple extension of SystemTestsCompareTwo, so if LII has stopped working in this respect, then presumably other tests like REP (simple reproducibility test) have started having the same problem.

@ekluzek - if you still have this info, can you please document your exact procedure as well as the way you're determining that the first case wasn't rerun?

**Comment by ekluzek** [comment]
The case I setup was:

LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way

I made changes that should have got it to work, but it was still telling me it failed. When I reran the entire test from scratch it worked.

The other thing I saw (or thought I saw) was that the "*.nc.interp" files had an older date than the "*.nc.base" files so it looked like to me they weren't being rerun.

Here's a listing of the run directory:

-rw-r--r-- 1 erik cgdtss  44133948 Apr 24 00:11 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.clm2.h0.0001-01-04-00000.nc.base
-rw-r--r-- 1 erik cgdtss  99497636 Apr 24 00:11 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.clm2.h1.0001-01-04-00000.nc.base
-rw-r--r-- 1 erik cgdtss 107085944 Apr 24 00:11 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.cpl.hi.0001-01-04-00000.nc.base
[erik@yslogin3 LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel]$ ls -ltard $MYSCR/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/*.nc.interp
lrwxrwxrwx 1 erik cgdtss 275 Apr 22 20:18 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.cpl.hi.0001-01-04-00000.nc.interp -> /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp.cpl.hi.0001-01-04-00000.nc.interp
lrwxrwxrwx 1 erik cgdtss 276 Apr 22 20:18 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.clm2.h1.0001-01-04-00000.nc.interp -> /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp.clm2.h1.0001-01-04-00000.nc.interp
lrwxrwxrwx 1 erik cgdtss 276 Apr 22 20:18 /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.clm2.h0.0001-01-04-00000.nc.interp -> /glade/scratch/erik/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp/run/LII_D_Ld3.f19_g16_gl4.IG1850CLM45.yellowstone_intel.clm-glcMEC_spunup_1way.GC.latestfil03_r23345intel.interp.clm2.h0.0001-01-04-00000.nc.interp

Note the "base" files have a Apr/24th date while the "interp" files have a Apr/22nd date. However, I then noticed that the files with the older date are the softlinks to the actual files which DO have a Apr/24th date.

So my initial assessment that it isn't rerunning one of the tests isn't correct. I think I should try doing this with cime master and see if it fails.

**Comment by jedwards4b** [comment]
I think that this is fixed on master with PR #1394 - can you retry with the latest cime?

**Comment by ekluzek** [comment]
OK. Bill and I talked this one over, and figured out what the problem was. As already stated my original analysis of the problem was incorrect. The problem was that I changed user_nl_clm in the parent case -- but didn't change it in the cloned case ".interp". If you change any of the settings in one case you also have to change it in the cloned case. So I'm just going to close this as "invalid" and "wontfix".
